### PR TITLE
CUDA documentation enhancement

### PR DESCRIPTION
**Pull request by elistevens**
Adds documentation that would have been helpful for me as I tried to navigate how to do multi-GPU training for the first time. Since I'm still learning the library, I'm not certain that the text I've added is correct! Please let me know if it could be improved and I'll incorporate the changes.

Also has some smaller tweaks for formatting issues, etc. that I noticed.

### CODE REVIEW

File: torch/serialization.py [hunk 1]
```diff
@@ -103,6 +103,29 @@ def storage_to_tensor_type(storage):
 def save(obj, f, pickle_module=pickle, pickle_protocol=DEFAULT_PROTOCOL):
     """Saves an object to a disk file.
 
+    There are two main approaches for serializing and restoring a model.
```
Comment (soumith) [inline torch/serialization.py]: model -> neural network model

Also, this blob has to be moved to notes, "Serialization Best Practices", and a reference to the notes can be inserted here in the reference docs (reference docs are not full notes).

File: docs/source/notes/cuda.rst [hunk 2]
```diff
@@ -61,3 +63,19 @@ call. This can be used to overlap data transfers with computation.
...
+
+.. _cuda-nn-dataparallel-instead:
+
+Use nn.DataParallel instead of multiprocessing
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Most use cases involving batched input and multiple GPUs should default to using
+:class:`~torch.nn.DataParallel` to utilize more than one GPU. Even with the GIL,
+a single python process can saturate multiple GPUs, though at very large numbers
+of GPUs (8+) utilization might drop. Test your use case before investing the
```
Comment (soumith) [inline docs/source/notes/cuda.rst]: Sam is working on the 8+ GPU case, in the next major release, utilization wouldn't drop in this case either, just fyi on what's coming next.

**Comment by soumith** [comment]
requested small changes via inline comments. otherwise, looks great :)

Also, pep8 lint is failing (minor whitespace fixes needed): https://travis-ci.org/pytorch/pytorch/jobs/205199910

File: torch/serialization.py (hunk 1 above)
Comment (apaszke) [inline torch/serialization.py]: I'd say that just "model" is better. NNs are special cases of models in general 😉

**Comment by soumith** [comment]
thanks!

**Comment by elistevens** [comment]
Oh, I was planning on doing the requested tweaks (moving things to serialization.rst, etc.) today. Probably should have said that, but yesterday was pretty busy.

Shall I submit another PR?

**Comment by apaszke** [comment]
Yes please!

**Comment by elistevens** [comment]
For things like this, are single-commit PRs preferred, or should I be letting maintainers do the rebase?

**Comment by soumith** [comment]
we'll do the rebase / squash  if needed.

File: torch/serialization.py (hunk 1 above)
Comment (elistevens) [inline torch/serialization.py]: Would it make sense to specify "a subclass of `nn.Model`"?

Comment (apaszke) [inline torch/serialization.py]: No! People that use the functional style or serialized other stuff should use it to! It has nothing to do with mm.

**Pull request merged**